core/file/dicom/image.cpp

```diff
@@ -35,6 +35,9 @@ namespace MR {
           // ignore anything within IconImageSequence:
           if (seq.group ==  0x0088U && seq.element == 0x0200U)
             return;
+          // ignore anything within sequences with unknown (private) group:
+          if (seq.group & 1U)
+            return;
```

Comment:
Does your DICOM test suite detect novel omission of header key-values? I can see theoretically that this change could lead to unintended loss of header information, even if that information was coming from a descendent of a private group, which wouldn't be ideal from a data standpoint. Hopefully there are no cases of such in your private test data and it can therefore be written off as a non-issue.

Reply:
That's a good point, but I think everything checks out on that front too on all the test data I have. The actual testing bits of the script I use are performed as:
```
    { 
      echo "${image%.*}" | mrconvert ../../"$folder" $TMPFILE 2>/dev/null && 
      testing_diff_image $TMPFILE "$image" -frac 1e-6 -quiet && 
      testing_diff_header $TMPFILE "$image" -keyval -quiet &&
      echo OK
    }
```
The first `echo` is simply the [DICOM selection](https://mrtrix.readthedocs.io/en/latest/tips_and_tricks/dicom_handling.html#piping-known-selections-to-the-command). The DICOM gets converted to `.mif`, then tested against the corresponding previously validated image for differences in image intensity (to within 6 sig. figs), and then tested for any differences in header entries, with the `-keyval` option making sure all entries are included in the check. 

I just checked that all of that works as expected: any property not found in both source and destination will raise an error, and any mismatch in the exact values stored in those fields will also trigger a mismatch. 

If anything, this means I'll probably end up having to update the validated data every so often as we decide to parse more and more DICOM fields...